branch/transformer/transformer.cc

```diff
@@ -1,62 +1,98 @@
+#include "transformer.hh"
+
 #include <algorithm>
 #include <array>
 #include <bitset>
 #include <cmath>
 #include <deque>
 #include <map>
+#include <random>
 
 #include "msl/fwcounter.h"
 #include "ooo_cpu.h"
+#include "FixedVector.hh"
 
 constexpr std::size_t WEIGHT_BITS = 8;     // We can quantize down to 4 later
 constexpr std::size_t HISTORY_LENGTH = 24; // We can adjust. Defaulting to current perceptron's length for closer 1-to-1 comparison
 
+
 namespace
 {
 template <std::size_t HISTLEN, std::size_t BITS> // We set the history length and size of weights
-class transformer
+class Transformer : public TransformerBase
 {
-private:
- // @deprecated
-  uint32_t cyclic_positional_encoding(uint32_t input_vector)
-  {
-    /*
-        Creates the appended positional encoding
+public:
+  Transformer(const std::string& config_file) : TransformerBase(config_file) {}
+
+  void hashed_posEncoding(uint64_t& input, std::bitset<HISTLEN> global_history){
```

Comment:
hashed_pos_encoding to keep it following the same style format as everything else?